specify/specify7
# Scope all front-end requests where needed

Fixes #2292

To test:
- Attachment counts are scoped to the current collection (can be checked by making sure the attachment count matches the number of displayed attachments in the attachment viewer) (they were not scoped before)
- Pick lists of type "Entire table" or "Single Field" have their pick list items scoped (they were not scoped before, @grantfitzsimmons do you see a problem with this change?)
